vector/README.md

```diff
@@ -1,6 +1,72 @@
 The `vector` package [![Build Status](https://github.com/haskell/vector/workflows/CI/badge.svg)](https://github.com/haskell/vector/actions?query=branch%3Amaster)
 ====================
 
-An efficient implementation of `Int`-indexed arrays (both mutable and immutable), with a powerful loop optimisation framework.
+This package includes various modules that will allow you to 
+work with vectors and use an optimisation framework called [*fusion*](#fusion). 
+In this context, vector is an `Int`-indexed array-like data structure with a simpler 
+API that can contain any Haskell value. Additionally, its equivalence 
+to C-style arrays and optimisation via fusion accelerates vector’s 
+performance and makes it a great alternative to list. By installing this 
+package, you’ll be able to work with [boxed, unboxed, storable, and primitive 
+vectors](#vectors-available-in-the-package) as well as their generic interface.
 
-See [`vector` on Hackage](http://hackage.haskell.org/package/vector) for more information.
+
+## Table of Contents
+
+<!-- no toc -->
+- [Tutorial](#tutorial)
+- [Vector vs Array](#vector-vs-array)
+- [Vectors Available in the Package](#vectors-available-in-the-package)
+- [Stream Fusion](#stream-fusion)
+
+## Tutorial
+
+A beginner-friendly tutorial for vectors can be found on 
+[MMHaskell](https://mmhaskell.com/data-structures/vector).
+
+
+If you have already started your adventure with vectors, 
+the tutorial on [Haskell Wiki](https://wiki.haskell.org/Numeric_Haskell:_A_Vector_Tutorial) 
+covers more ground.
+
+## Vector vs Array
```

Comment:
Do we even need to compare with `array`. I think at this point it's mostly legacy and barely used

Reply:
we can remove it, but I think we should keep it for the newcomers. If array is legacy and rarely used, I think it's important to mention that so I added a sentence